test-fixtures/src/bin/request.rs

```diff
@@ -420,11 +421,31 @@ fn test_header_multi_value_set_and_get() {
     assert_eq!(0, no_header_names.len());
 }
 
+fn test_default_decompress_response() {
+    let mut request = FastlyRequestHandle::new();
+
+    request.set_auto_decompress_response(ContentEncodings::default());
+}
+
+fn test_gzip_decompress_response() {
+
+    let result = panic::catch_unwind(|| {
+        let mut request = FastlyRequestHandle::new();
+
+        request.set_auto_decompress_response(ContentEncodings::GZIP);
+    });
+    if result.is_ok() {
+        panic!("Expected `request.set_auto_decompress_response(ContentEncodings::GZIP);` to cause a panic")
+    }
+}
```

Comment:
This won't work because wasm32-unknown-unknown is panic="abort" by default.

@cratelyn - do you know of a way that we can test panic behavior within the wasm binaries?

Reply:
If we can't test panic behavior then we can remove this test for the change.
This commit shows that the other changes and tests in the pull-request do pass on CI -- [`3166b54` (#116)](https://github.com/fastly/Viceroy/pull/116/commits/3166b54860433add1ba7e1cd607069f5ee6b35ba)

Reply:
> @cratelyn - do you know of a way that we can test panic behavior within the wasm binaries?

short answer: not _really_

slightly more involved answer: yes, but i don't believe that it would be worth the effort in this particular case. we have a fixture called `trap-test`, which is used to check that a panic is handled correctly. _that said_, it is an involved process involving conditional compilation in `viceroy-lib` and a test fixture that must be compiled separately from the other test fixtures.

if you're curious, `trap-test` and `test-fatalerror-config` are two phrases that will yield interesting result when grepping around the repository!

anyways, **tldr;,** we can remove this test; hopefully, we'll implement that feature soon.

Reply:
Thank you @cratelyn ☺️ - I will have a read through as I am curious.

I will also remove this test so that this pull-request can pass the tests on CI 👍

Reply:
:cat: wonderful! we should also be sure to remove the `panic = "unwind"` profiles from `test-fixtures/Cargo.toml`, and then this should be in good shape. thank you for putting this together @JakeChampion! :trophy: